lib/populator.js

```diff
@@ -79,8 +79,12 @@ Populator.prototype._getTrackedEntityAttributeType = function (trackedEntityAttr
       return next(new Error('Unexpected status code ' + res.statusCode));
     }
     this._cache.trackedEntityAttributeTypes[trackedEntityAttributeID] = body.valueType;
-    if (!this._cache.firstUniqueTrackedEntityAttributeID && body.unique === true) {
-      this._cache.firstUniqueTrackedEntityAttributeID = trackedEntityAttributeID;
+    if (this._options.uniqueAttributeID && !this._cache.uniqueTrackedEntityAttributeID) {
+      this._cache.uniqueTrackedEntityAttributeID = this._options.uniqueAttributeID;
```

Comment:
@MattyJ007 Can this initialisation not be done in the constructor after the cache has been constructed instead? Around https://github.com/jembi/dhis2-tracker-populator/blob/e85fd787c78ba0887322cf7e4c5aa4ab93f65f4b/lib/populator.js#L16.